libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/SessionCredentialItem.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.di.ipv.core.library.persistence.item;
+
+import com.nimbusds.jwt.SignedJWT;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSortKey;
+import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
+
+@ExcludeFromGeneratedCoverageReport
+@DynamoDbBean
+public class SessionCredentialItem implements DynamodbItem {
+
+    private static final String SORT_KEY_TEMPLATE = "%s#%s";
+    private final String ipvSessionId;
+    private final String sortKey;
+    private final String credential;
+    private final boolean receivedThisSession;
+    private long ttl;
+
+    public SessionCredentialItem(
+            String ipvSessionId,
+            String criId,
+            SignedJWT signedCredJwt,
+            boolean receivedThisSession) {
+        this.ipvSessionId = ipvSessionId;
+        this.sortKey = String.format(SORT_KEY_TEMPLATE, criId, signedCredJwt.getSignature());
```

Comment:
Do you think it'd be useful to include the reverse, i.e. so there is a `getCriId` method? You could also consider making the private field `criId`, and only generate the sort key on-demand in `getSortKey` - that might simplify the constructor.

(The only thing to watch out for is I'm not quite sure how `@DynamoDbBean` selects the fields to write into the record...)

Reply:
If this is tricky to work out now, you could also leave it to the next ticket.

Reply:
I was mostly aiming for the minimum here to express what the sort key should be - there's a good chance stuff will move around in the next ticket. Will bear this in mind.